includes/Core/Site_Health/Tags_Placement.php

```diff
@@ -0,0 +1,279 @@
+
+		if ( is_wp_error( $response ) ) {
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				__( 'There was an error while trying to get the status, please try again later.', 'google-site-kit' )
+			);
+
+			return $result;
+		}
+
+		$response = wp_remote_retrieve_body( $response );
+
+		$description = array();
+		foreach ( $active_modules as $module ) {
+			$tag_found = $this->check_if_tag_exists( $module, $response );
+
+			if ( $tag_found ) {
+				$description[] = $tag_found;
+			}
+		}
+
+		if ( ! empty( $description ) ) {
+			$result['description'] = '<ul>' . join( "\n", $description ) . '</ul>';
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Gets active modules filtered to account only for
+	 * Analytics, AdSense and Tag Manager.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Filtered active modules instances.
+	 */
+	protected function get_active_modules() {
+		$active_modules = $this->modules->get_active_modules();
+
+		$active_modules = array_filter(
+			$active_modules,
+			function( $module ) {
+				return $module instanceof Module_With_Tag;
+			}
+		);
+
+		return $active_modules;
+	}
+
+	/**
+	 * Checks if tag exists.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Module_With_Tag $module  Module instance.
+	 * @param string          $content Content to search for the tags.
+	 * @return bool TRUE if tag is found, FALSE if not.
+	 */
+	protected function check_if_tag_exists( $module, $content ) {
+		$check_tag   = $module->has_placed_tag_on_frontend( $content );
+		$module_name = $this->get_module_name_from_slug( $module->slug );
+
+		switch ( $check_tag ) {
+			case Module_Tag_Matchers::TAG_EXISTS_WITH_COMMENTS:
+				return sprintf(
+					'<li><strong>%s</strong>: %s</li>',
+					$module_name,
+					__( 'Tag detected and placed by Site Kit.', 'google-site-kit' )
+				);
+
+			case Module_Tag_Matchers::TAG_EXISTS:
+				return sprintf(
+					'<li><strong>%s</strong>: %s</li>',
+					$module_name,
+					__( 'Tag detected but could not verify that Site Kit placed the tag.', 'google-site-kit' )
+				);
+
+			case Module_Tag_Matchers::NO_TAG_FOUND:
+				return sprintf(
+					'<li><strong>%s</strong>: %s</li>',
+					$module_name,
+					__( 'No tag detected.', 'google-site-kit' )
+				);
+
+			default:
+				return sprintf(
+					'<li><strong>%s</strong>: %s</li>',
+					$module_name,
+					__( 'No tag detected.', 'google-site-kit' )
+				);
+		}
+	}
+
+	/**
+	 * Returns formated module string name.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $slug Module slug.
+	 * @return string Formatted module name.
+	 */
+	public function get_module_name_from_slug( $slug ) {
+		if ( 'adsense' === $slug ) {
+			$module_name = 'AdSense';
+		}
+		if ( 'analytics-4' === $slug ) {
+			$module_name = 'Analytics';
+		}
+		if ( 'tagmanager' === $slug ) {
+			$module_name = 'Tag Manager';
+		}
+
+		return $module_name;
+	}
```

Comment:
This method shouldn't be here. Let's probably add a new public abstract method `get_public_name` to the parent Module class that should be implemented in all children modules and then use that method in this class to get the public name of the module.